### PR TITLE
SONAR-6729 Analysis should fail when no quality profile

### PR DESCRIPTION
**Pull request by julienlancelot**
SONAR-6729 Analysis should fail when no quality profile

### CODE REVIEW

File: sonar-batch/src/main/java/org/sonar/batch/repository/DefaultProjectRepositoriesLoader.java [hunk 1]
```diff
@@ -52,7 +53,16 @@ public ProjectRepositories load(ProjectReactor reactor, AnalysisProperties taskP
...
-    return ProjectRepositories.fromJson(serverClient.request(url));
+
+    ProjectRepositories projectRepositories = ProjectRepositories.fromJson(serverClient.request(url));
+    validateProjectRepositories(projectRepositories, reactor.getRoot().getKey());
+    return projectRepositories;
+  }
+
+  private void validateProjectRepositories(ProjectRepositories projectRepositories, String projectKey) {
+    if (projectRepositories.qProfiles().isEmpty()) {
+      throw MessageException.of("No quality profile has been found on this project");
```
Comment (henryju) [inline sonar-batch/src/main/java/org/sonar/batch/repository/DefaultProjectRepositoriesLoader.java]: Should it be "No quality profiles have been found..."? @ganncamp help :)

File: sonar-batch/src/main/java/org/sonar/batch/repository/DefaultProjectRepositoriesLoader.java [hunk 2]
```diff
@@ -52,7 +53,16 @@ public ProjectRepositories load(ProjectReactor reactor, AnalysisProperties taskP
...
     }
     url += "&preview=" + analysisMode.isPreview();
-    return ProjectRepositories.fromJson(serverClient.request(url));
+
+    ProjectRepositories projectRepositories = ProjectRepositories.fromJson(serverClient.request(url));
+    validateProjectRepositories(projectRepositories, reactor.getRoot().getKey());
+    return projectRepositories;
+  }
+
+  private void validateProjectRepositories(ProjectRepositories projectRepositories, String projectKey) {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/repository/DefaultProjectRepositoriesLoader.java]: MAJOR Make "validateProjectRepositories" a "static" method. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: sonar-batch/src/main/java/org/sonar/batch/repository/DefaultProjectRepositoriesLoader.java (hunk 1 above)
Comment (julienlancelot) [inline sonar-batch/src/main/java/org/sonar/batch/repository/DefaultProjectRepositoriesLoader.java]: Hum, probably you're right !

**Comment by sns-seb** [comment]
@julienlancelot back to you, with some questions?

**Comment by julienlancelot** [comment]
@sns-seb I've answered your questions :)

**Comment by julienlancelot** [comment]
@sns-seb Feedback applied

**Comment by sns-seb** [comment]
b9ilw-5ciaethav

**Pull request merged**